src/codemagic/tools/app_store_connect/action_groups/certificates_action_group.py

```diff
@@ -0,0 +1,228 @@
+    ) -> None:
+        """
+        Delete specified Signing Certificate from Apple Developer portal
+        """
+
+        self._delete_resource(self.api_client.signing_certificates, certificate_resource_id, ignore_not_found)
+
+    @cli.action(
+        "list",
+        CertificateArgument.CERTIFICATE_TYPES_OPTIONAL,
+        CertificateArgument.PROFILE_TYPE_OPTIONAL,
+        CertificateArgument.DISPLAY_NAME,
+        CertificateArgument.PRIVATE_KEY,
+        CertificateArgument.PRIVATE_KEY_PASSWORD,
+        CertificateArgument.P12_CONTAINER_PASSWORD,
+        CommonArgument.SAVE,
+        action_group=AppStoreConnectActionGroup.CERTIFICATES,
+        deprecation_info=cli.ActionDeprecationInfo("list-certificates", "0.49.0"),
+    )
+    def list_certificates(
+        self,
+        certificate_types: Optional[Union[CertificateType, Sequence[CertificateType]]] = None,
+        profile_type: Optional[ProfileType] = None,
+        display_name: Optional[str] = None,
+        certificate_key: Optional[Union[PrivateKey, Types.CertificateKeyArgument]] = None,
+        certificate_key_password: Optional[Types.CertificateKeyPasswordArgument] = None,
+        p12_container_password: str = "",
+        save: bool = False,
+        should_print: bool = True,
+        **_deprecated_kwargs,
+    ) -> List[SigningCertificate]:
+        """
+        List Signing Certificates from Apple Developer Portal matching given constraints
+        """
+
+        private_key = self._get_certificate_key(certificate_key, certificate_key_password)
+        if save and private_key is None:
+            raise AppStoreConnectError("Cannot create or save resource without certificate private key")
+
+        _certificate_type: Optional[CertificateType] = _deprecated_kwargs.get("certificate_type")
+        certificate_types_filter = self._resolve_certificate_types(_certificate_type, certificate_types, profile_type)
+
+        certificate_filter = self.api_client.signing_certificates.Filter(
+            certificate_type=certificate_types_filter,
+            display_name=display_name,
+        )
+        certificates = self._list_resources(
+            certificate_filter,
+            self.api_client.signing_certificates,
+            should_print,
+        )
+
+        if private_key:
+            certificates = [
+                certificate
+                for certificate in certificates
+                if Certificate.from_ans1(certificate.asn1_content).is_signed_with(private_key)
+            ]
+            self.printer.log_filtered(SigningCertificate, certificates, "for given private key")
+            for certificate in certificates:
+                self.logger.info(f"- {certificate.get_display_info()}")
+
+        if save:
+            self._save_certificates(
+                certificates,
+                cast(PrivateKey, private_key),
+                p12_container_password,
+            )
+
+        return certificates
+
+    def _resolve_certificate_types(
```

Comment:
Used to be defined in https://github.com/codemagic-ci-cd/cli-tools/blob/3670fa624997f17cb437bc88198f7bdee3594f35/src/codemagic/tools/app_store_connect.py#L627-L664